sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -3,479 +3,624 @@
+  ? {}
+  : never;
+
+// @public
+export interface ContextWithSettlement {
+  abandon(
+    message: ReceivedMessage,
+    propertiesToModify?: {
+      [key: string]: any;
+    }
+  ): Promise<void>;
+  complete(message: ReceivedMessage): Promise<void>;
+  deadLetter(message: ReceivedMessage, options?: DeadLetterOptions): Promise<void>;
+  defer(
+    message: ReceivedMessage,
+    propertiesToModify?: {
+      [key: string]: any;
+    }
+  ): Promise<void>;
+}
+
 // @public
 export interface CorrelationFilter {
-    contentType?: string;
-    correlationId?: string;
-    label?: string;
-    messageId?: string;
-    replyTo?: string;
-    replyToSessionId?: string;
-    sessionId?: string;
-    to?: string;
-    userProperties?: any;
+  contentType?: string;
+  correlationId?: string;
+  label?: string;
+  messageId?: string;
+  replyTo?: string;
+  replyToSessionId?: string;
+  sessionId?: string;
+  to?: string;
+  userProperties?: any;
 }
 
-export { DataTransformer }
+// @public
+export interface CreateSessionReceiverOptions extends OperationOptions {
+  maxSessionAutoRenewLockDurationInSeconds?: number;
+}
+
+export { DataTransformer };
 
 // @public
 export interface DeadLetterOptions {
-    deadLetterErrorDescription: string;
-    deadletterReason: string;
+  deadLetterErrorDescription: string;
+  deadletterReason: string;
 }
 
-export { delay }
+export { delay };
+
+export { Delivery };
 
-export { Delivery }
+// @public
+export type EntityStatus =
+  | "Active"
+  | "Creating"
+  | "Deleting"
+  | "ReceiveDisabled"
+  | "SendDisabled"
+  | "Disabled"
+  | "Renaming"
+  | "Restoring"
+  | "Unknown";
 
 // @public
-export type EntityStatus = "Active" | "Creating" | "Deleting" | "ReceiveDisabled" | "SendDisabled" | "Disabled" | "Renaming" | "Restoring" | "Unknown";
+export interface GetMessageIteratorOptions extends OperationOptions {}
 
-export { HttpOperationResponse }
+export { HttpOperationResponse };
+
+// @public
+export interface MessageAndContext<ContextT> {
+  context: ContextT;
+  message: ReceivedMessage;
+}
 
 // @public
 export type MessageCountDetails = {
-    activeMessageCount: number;
-    deadLetterMessageCount: number;
-    scheduledMessageCount: number;
-    transferMessageCount: number;
-    transferDeadLetterMessageCount: number;
+  activeMessageCount: number;
+  deadLetterMessageCount: number;
+  scheduledMessageCount: number;
+  transferMessageCount: number;
+  transferDeadLetterMessageCount: number;
 };
 
 // @public
 export interface MessageHandlerOptions {
-    autoComplete?: boolean;
-    maxConcurrentCalls?: number;
-    maxMessageAutoRenewLockDurationInSeconds?: number;
+  autoComplete?: boolean;
+  maxConcurrentCalls?: number;
+  maxMessageAutoRenewLockDurationInSeconds?: number;
 }
 
-export { MessagingError }
+// @public
+export interface MessageHandlers<ContextT> {
+  processError(err: Error): Promise<void>;
+  processMessage(message: ReceivedMessage, context: ContextT): Promise<void>;
```

Comment:
I'm only handing the user one at a time. Should we give them a batch of methods? 

Is the use case for service bus different than EventHubs, for example?